packages/@aws-cdk/aws-batch/lib/job-definition.ts

```diff
@@ -46,6 +53,28 @@ export enum LogDriver {
   SYSLOG = 'syslog'
 }
 
+/**
+ * An object representing the secret to expose to your container
+ */
+export interface SecretProperty {
+  /**
+   * The name of the secret.
+   */
+  readonly name: string;
```

Comment:
Im wondering about this. Is this a name the user makes up or should this be the name of the existing secret being referenced in the `value` below?

Reply:
I think since the value prop is named `ValueFrom`, the name is user driven and they make it up.

Reply:
I think that this name is the option name being configured on the docker daemon as the `log-opts` property . So it's actually not made up but should match the option that the underlying log driver expects. 

https://docs.aws.amazon.com/batch/latest/userguide/specifying-sensitive-data.html

> Secrets can be exposed to a job in the following ways:
>
> - To inject sensitive data into your containers as environment variables, use the secrets job definition parameter.
>
> - To reference sensitive information in the log configuration of a job, use the secretOptions job definition parameter.

For example when configuring the `SPLUNK` log driver, it has to be *splunk-token*.

https://docs.docker.com/config/containers/logging/splunk/

```json
{
  "log-driver": "splunk",
  "log-opts": {
    "splunk-token": "",
    "splunk-url": "",
    ...
  }
}
```

Would be amazing if in the future we can codify this in the API. Something like:

```ts
logDriver: batch.LogDriver.splunk(...) // internally this would configure the 'splunk-token' name.
```

But for now, lets just rename to `optionName` and give a brief explanation and link.

Reply:
@iliapolo These properties are not part of secrets. They are part of the `options` prop that we are asking. These `options` are configuration options specific to the selected log driver. The secret (and its name) is totally independent of it.

Reply:
I'm not sure thats correct. In that case what is the meaning of this name? How will the secret value be used in the container? 

What i believe is happening is that CloudFormation (Or batch) merges the `options` and the `secrets` to the same options map, and passes it to the docker daemon. The reason they are separate configs is to enable specifying secret arns in a more modeled way.

See example here: https://docs.aws.amazon.com/batch/latest/userguide/specifying-sensitive-data-secrets.html#secrets-logconfig

I don't think its accidental that they use *splunk-token* as the secret name.

Reply:
Aah I see. I understand it now. Thanks for clarifying this in detail.

Let me make the required changes.